zerver/lib/bugdown/__init__.py

```diff
@@ -1763,6 +1763,40 @@ def get_sub_registry(r: markdown.util.Registry, keys: List[str]) -> markdown.uti
 DEFAULT_BUGDOWN_KEY = -1
 ZEPHYR_MIRROR_BUGDOWN_KEY = -2
 
+class NormalizeWhitespace(markdown.preprocessors.NormalizeWhitespace):
+    # This class doesn't remove start and end of text characters from text.
+    # This is to allow HtmlStash placeholders to function properly on
+    # fenced codeblocks we find inside block level elements.
+
+    def run(self, lines: List[str]) -> List[str]:
+        source = '\n'.join(lines)
+        # source = source.replace(util.STX, "").replace(util.ETX, "")
```

Comment:
Can you document more clearly what you've changed from upstream?  E.g. I assume this is the original upstream code; if so, we should say that.

More generally, I'd like to understand the reasoning for why this won't create other issues.